components/forms/Donate/Donatemonthly.tsx

```diff
@@ -11,14 +11,14 @@ const stepFour = ' covers the cost of four therapy sessions each month.'
 
 export default function Donatemonthly() {
   const [loading, setLoading] = useState(false)
-  const [input, setInput] = useState()
+  const [input, setInput] = useState({ value: 0 })
   const [impact, setImpact] = useState('$30')
   const [message, setMessage] = useState(stepTwo)
 
-  const handleInputChange = (e) => {
-    const id = e.target.id
-    const value = e.target.value
-    const provides = e.target.step
+  const handleInputChange = (e: FormEvent<HTMLInputElement>) => {
+    const id = Number(e.currentTarget.id)
+    const value = Number(e.currentTarget.value)
```

Comment:
Is there a reason to go with the generic Number casting vs parseInt or parseFloat. I believe you'll always receive a string and stripe will probably either want an Int or a Float

Reply:
```suggestion
    const value = parseInt(e.currentTarget.value)
```